clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -3621,24 +3641,37 @@ public RemoveMembersFromConsumerGroupResult removeMembersFromConsumerGroup(Strin
         KafkaFutureImpl<Map<MemberIdentity, Errors>> future = new KafkaFutureImpl<>();
 
         ConsumerGroupOperationContext<Map<MemberIdentity, Errors>, RemoveMembersFromConsumerGroupOptions> context =
-            new ConsumerGroupOperationContext<>(groupId, options, deadline, future);
+                new ConsumerGroupOperationContext<>(groupId, options, deadline, future);
 
-        Call findCoordinatorCall = getFindCoordinatorCall(context,
-            () -> getRemoveMembersFromGroupCall(context));
+        Call findCoordinatorCall;
+        if (options.removeAll()) {
+            List<MemberIdentity> members = getMembersFromGroup(groupId);
+            findCoordinatorCall = getFindCoordinatorCall(context,
+                () -> getRemoveMembersFromGroupCall(context, members));
```

Comment:
My initial thought was to put the `members` in the context, but hesitated to do so because the `ConsumerGroupOperationContext` seems to be for generic usage. So I just refer to `KafkaAdminClient#getAlterConsumerGroupOffsetsCall` and make the members as a separate input param. Anyway, I'm glad to make the change if we think it's preferred to put the `members` in context.

Reply:
And to be clear, I'm not suggesting we have to put stuff into the context, just always passing in the intended removal list and do not depend on `context.removeAll` again inside internal function.